### DISCUSSION TITLE
Compare implementation supports of sections in [ ] like in .REG and .INF files

### DISCUSSION THREAD

**Discussion by ohault** [WinMerge/winmerge#1975] [Ideas]
When comparing text files with sections like the one in .REG and .INF with Windows, we can expect WinMerge being able to contextualize comparison not line by line in order, but as a mapping of all the line within a section.

The goal would be to avoid hereunder issue.  

![image](https://github.com/WinMerge/winmerge/assets/5320161/7c3ff599-57d9-4912-8737-efed4278adda)

Does it already possible to support this need in some ways or it could be a new feature ?

**Comment by sdottaka** [comment]
Unfortunately, WinMerge doesn't have the ability to compare this very well.

By right-clicking on the Location Pane as shown below and selecting the "All Moved Blocks" menu item, you can detect moved blocks. This might provide some assistance.

![image](https://github.com/WinMerge/winmerge/assets/98126/7fbd5d89-aced-4057-842d-deea5086aec0)

if you have a command-line tool that can sort INI or REG files and output the results to standard output, you can compare the sorted results by specifying arguments in the ExecFilterCommand plugin as follows:

~~~
ExecFilterCommand c:\tools\inisort %1
~~~

![image](https://github.com/WinMerge/winmerge/assets/98126/167bc60b-f0bf-4a9e-8185-849c26ab587f)

**Comment by ohault** [comment]
Thank you @sdottaka for your answer. I hope this feature will be integrated soon natively into WinMerge.
One can expect WinMerge should provide this feature by default by detecting such structured files automatically. In fact, from a functional perspective it's a design bug because it does not make sense to compare files with such file format structure line by line as flat files.

**Reply by ohault** [reply]
Any updates on this?
Using an INI file library could it be a good way to implement this feature in WinMerge ?
e.g. https://github.com/compuphase/minIni/issues/27
